lib/ex_force/oauth.ex

```diff
@@ -163,8 +163,13 @@ defmodule ExForce.OAuth do
       |> DateTime.to_unix(:millisecond)
       |> Integer.to_string()
 
-    :sha256
-    |> :crypto.hmac(client_secret, id <> issued_at_raw)
+    crypto_args = [:sha256, client_secret, id <> issued_at_raw]
+
+    if function_exported?(:crypto, :mac, 4) do
```

Comment:
nitpick - `function_exported?` will be called for every invocation of this call, and it can be avoided by moving it to outside of `def` - compile time.

Example:

```elixir
defmodule Foo do
  def inside(key, data) do
    if function_exported?(:crypto, :mac, 4) do
      :crypto.mac(:hmac, :sha256, key, data)
    else
      :crypto.hmac(:sha256, key, data)
    end
  end

  if function_exported?(:crypto, :mac, 4) do
    def outside(key, data) do
      :crypto.mac(:hmac, :sha256, key, data)
    end
  else
    def outside(key, data) do
      :crypto.hmac(:sha256, key, data)
    end
  end
end
```

Margin could be very small (my local benchee test says 1-2% slower) - but this is so small so let's do it. Let's define a private func and use it.

Reply:
sorry for the delay

I've updated the code with how [plug_crypto](https://github.com/elixir-plug/plug_crypto/blob/7d6923b3f967f574774b1fe322c511f9fa2fb3e6/lib/plug/crypto/key_generator.ex#L96-L100) is handling this